packages/preferences/src/common/preference-service.ts

```diff
@@ -32,60 +34,70 @@ export class PreferenceService implements Disposable {
         this.toDispose.dispose();
     }
 
+    ready(): Promise<void> {
```

Comment:
How should clients check for preference service readiness then? With a get to the promise?